templates/typescript_gapic_metadata/src/$version/gapic_metadata.json.njk

```
@@ -0,0 +1,103 @@
+  has no idea about TypeScript): all the naming logic is in the templates.
+  Hence, the metadata JSON is also a template that uses the same naming
+  logic. We'll do our best to make it compatible with the proper schema.
+-#}
+
+{% import "../../_namer.njk" as namer -%}
+{{- namer.initialize(id, api) -}}
+
+{
+  "schema": "1.0",
+  "comment": "This file maps proto services/RPCs to the corresponding library clients/methods",
+  "language": "typescript",
+  "protoPackage": "{{ api.naming.protoPackage }}",
+  "libraryPackage": "{{ api.publishName }}",
+  "services": {
+{%- set serviceJoiner = joiner(',') -%}
+{%- for service in api.services -%}
+    {{- serviceJoiner() -}}
+    "{{ service.name }}": {
+      "grpc": {
+        "libraryClient": "{{ service.name }}Client",
+        "rpcs": {
+{%- set grpcMethodJoiner = joiner(',') -%}
+{%- for method in service.simpleMethods -%}
+          {{- grpcMethodJoiner() -}}
+          "{{ method.name }}": ["{{ method.name.toCamelCase() }}"]
+{%- endfor -%}
+{%- for method in service.streaming -%}
+          {{- grpcMethodJoiner() -}}
+          "{{ method.name }}": ["{{ method.name.toCamelCase() }}"]
+{%- endfor -%}
+{%- for method in service.longRunning -%}
+          {{- grpcMethodJoiner() -}}
+          "{{ method.name }}": [
+                                "{{ method.name.toCamelCase() }}",
+                                "{{ id.get("check" + method.name.toPascalCase() + "Progress", service.name) }}"
+                               ]
+{%- endfor -%}
+{%- for method in service.paging %}
+          {{- grpcMethodJoiner() -}}
+          "{{ method.name }}": [
+                                "{{ method.name.toCamelCase() }}",
+                                "{{ id.get(method.name.toCamelCase() + "Stream", service.name) }}",
+                                "{{ id.get(method.name.toCamelCase() + "Async", service.name) }}"
+                               ]
+{%- endfor -%}
+        }
+      },
+      "proto-over-http": {
```

nit: Is this the standard name we want to use? If other generators implement this as well, they should use the same name for the same transport (and we should codify it somewhere). I only ask because I'd personally prefer if it could be made shorter and yet self-describing (maybe `http-proto`?), but that's a distant second to being consistent with convention/existing uses.

---

I just invented this name and I'm open to using any other name (I was thinking of `"grpc-fallback"` but it's misleading because it neither says "http" nor "proto"). `http-proto` is good too!

---

Changed to `grpc-fallback` as discussed.